### PR TITLE
[GLUTEN-4039][VL] Add array forall and exists function support

### PR DESCRIPTION
**Pull request by lyy-pineapple**
## What changes were proposed in this pull request?

Add partial support of `forall` and `exists(array)` function for velox backend. This PR leverages existing all_match and any_match function in velox which supports forall with lambda function<T, boolean>.

## How was this patch tested?

Add uts

### CODE REVIEW

**Comment by lyy-pineapple** [comment]
Could you please review this PR, @PHILO-HE  ? Thanks

**Comment by lyy-pineapple** [comment]
@PHILO-HE Would you have time to help review this patch?

File: shims/common/src/main/scala/org/apache/gluten/expression/ExpressionNames.scala [hunk 1]
```diff
@@ -246,6 +246,8 @@ object ExpressionNames {
   final val ARRAY_REPEAT = "array_repeat"
   final val ARRAY_REMOVE = "array_remove"
   final val FILTER = "filter"
+  final val ALL_MATCH = "all_match"
+  final val ANY_MATCH = "any_match"
```
Comment (Yohahaha) [inline shims/common/src/main/scala/org/apache/gluten/expression/ExpressionNames.scala:250]: shims module is common for all backends, we should use vanilla spark's function name here, and do conversion in each backends internal if needed.

so, I expect here should be `forall` and `exist`.

File: gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionMappings.scala [hunk 2]
```diff
@@ -230,6 +230,8 @@ object ExpressionMappings {
     Sig[ArrayRepeat](ARRAY_REPEAT),
     Sig[ArrayRemove](ARRAY_REMOVE),
     Sig[ArrayFilter](FILTER),
+    Sig[ArrayForAll](ALL_MATCH),
+    Sig[ArrayExists](ANY_MATCH),
```
Comment (Yohahaha) [inline gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionMappings.scala:234]: same.

File: shims/common/src/main/scala/org/apache/gluten/expression/ExpressionNames.scala (hunk 1 above)
Comment (lyy-pineapple) [reply shims/common/src/main/scala/org/apache/gluten/expression/ExpressionNames.scala]: Fixed

File: gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionMappings.scala (hunk 2 above)
Comment (lyy-pineapple) [reply gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionMappings.scala]: Done

**Review by Yohahaha — APPROVED** [review]
LGTM! please fix conflicts and wait all CI passed.

**Comment by lyy-pineapple** [comment]
> LGTM! please fix conflicts and wait all CI passed.

has solved conflict ~ thanks

**Comment by PHILO-HE** [comment]
> @PHILO-HE Would you have time to help review this patch?

The pr looks good! Sorry for late response due to my time off in the last a few days.

**Pull request merged**